@xen-orchestra/rest-api/src/backups/jobs.service.mts

```diff
@@ -0,0 +1,22 @@
+import { BACKUP_TYPE, XoBackupJob, XoMetadataJob, XoMirrorJob } from '@vates/types'
+import { RestApi } from '../rest-api/rest-api.mjs'
+
+export class JobService {
+  #restApi: RestApi
+
+  constructor(restApi: RestApi) {
+    this.#restApi = restApi
+  }
+
+  async getVmJobs(): Promise<XoBackupJob[]> {
+    return this.#restApi.xoApp.getAllJobs(BACKUP_TYPE.backup).then(jobs => jobs.flat(1)) as Promise<XoBackupJob[]>
+  }
```

Comment:
By being more strict with the prop `type` in `XoBackupJob, XoMetadataJob, ...` and by exporting a new type `export type AnyXoBackupJob = XoBackupJob | XoMetadataJob | XoMirrorJob `you can enhance a lot your type, without the need of the `as` keyword.

```ts
// rest-api.type.ts

  getAllJobs(): Promise<AnyXoJob[]>
  getAllJobs<T extends AnyXoBackupJob['type']>(type: T): Promise<Extract<AnyXoBackupJob, { type: T }>[]>

...
// jobs.service.mts

const jobs = await this.#restApi.xoApp.getAllJobs() // typed as AnyXoJob[] 
const backupJobs = await this.#restApi.xoApp.getAllJobs('backup') // typed as XoBackupJob[]
const mirrorJobs = await this.#restApi.xoApp.getAllJobs('mirrorBackup') // typed as XoMirrorJob[] 

```